### PR TITLE
Update server.ts

### LINKED ISSUE #5433: settings: sdkPath/sdkVersion does not work
## Issue Description ##
settings: sdkPath/sdkVersion does not work.  
One workaround is to use global.json + "omnisharp.dotnetPath".  

## Steps to Reproduce ##
Use dotnet install from non-default location.  
Set "omnisharp.sdkVersion", "omnisharp.sdkPath" correctly - OmniSharp fails to start.  
Set "omnisharp.dotnetPath", "omnisharp.sdkVersion", "omnisharp.sdkPath" correctly - OmniSharp fails to start.  

## Expected Behavior ##
OmniSharp starts successfuly.

## Actual Behavior ##
OmniSharp fails to start.

The highest version SDK (7.0.100-rc.2.22477.23) has some problems, but that is why I set sdkVersion/sdkPath.  
Maybe OmniSharp server fails due to SDK defects, but I could not see why it do not starts with the specified version.  
(The problem is that with sdk=7.0 on my machine, dotnet build etc. mystically fails, but dotnet <MSBuild.dll ...> succeeds.)

### OmniSharp log ###
<details>
Starting OmniSharp server at 10/28/2022, 7:56:27 PM
    Target: [deleted]

OmniSharp server started with .NET 7.0.100-rc.2.22477.23
 (/home/ulp/.dotnet).
    Path: /home/ulp/.vscode/extensions/ms-dotnettools.csharp-1.25.0-linux-x64/.omnisharp/1.39.0-net6.0/OmniSharp.dll
    PID: 3738114

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on Unknown 0.0 (Unknown)
[info]: OmniSharp.Services.DotNetCliService
        Checking the 'DOTNET_ROOT' environment variable to find a .NET SDK
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to /home/ulp/.dotnet/dotnet
[fail]: OmniSharp.MSBuild.Discovery.Providers.SdkInstanceProvider
        The Sdk version specified in the OmniSharp settings was not a valid semantic version. Configured version is ''6.0.402''. Please update your settings and restart OmniSharp.
[fail]: OmniSharp.MSBuild.Discovery.Providers.SdkOverrideInstanceProvider
        The Sdk path specified in the OmniSharp settings does not exist. Configured path is ''/home/ulp/.dotnet/sdk/6.0.402''. Please update your settings and restart OmniSharp.
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 0 MSBuild instance(s)
Could not locate MSBuild instance to register with OmniSharp.
[ERROR] Error: OmniSharp server load timed out. Use the 'omnisharp.projectLoadTimeout' setting to override the default delay (one minute).
</details>
### C# log ###
<details>Post the output from Output-->C# here</details>

## Environment information ##

**VSCode version**: 1.72.2
**C# Extension**: 1.25.0

<details><summary>Mono Information</summary>
        OmniSharp using mono :6.12.0</details>
<details><summary>Dotnet Information</summary>
.NET SDK:
 Version:   7.0.100-rc.2.22477.23
 Commit:    0a5360315a

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /home/ulp/.dotnet/sdk/7.0.100-rc.2.22477.23/

Host:
  Version:      7.0.0-rc.2.22472.3
  Architecture: x64
  Commit:       550605cc93

.NET SDKs installed:
  6.0.402 [/home/ulp/.dotnet/sdk]
  7.0.100-rc.2.22477.23 [/home/ulp/.dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.10 [/home/ulp/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-rc.2.22476.2 [/home/ulp/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [/home/ulp/.dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.10 [/home/ulp/.dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-rc.2.22472.3 [/home/ulp/.dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/home/ulp/.dotnet]

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|cmake|twxs|0.0.17|
|cmake-tools|ms-vscode|1.12.27|
|cpptools|ms-vscode|1.12.4|
|cpptools-extension-pack|ms-vscode|1.3.0|
|cpptools-themes|ms-vscode|2.0.0|
|csharp|ms-dotnettools|1.25.0|
|intellicode-api-usage-examples|VisualStudioExptTeam|0.2.6|
|java|redhat|1.12.0|
|jupyter|ms-toolsai|2022.9.1202862440|
|jupyter-keymap|ms-toolsai|1.0.0|
|jupyter-renderers|ms-toolsai|1.0.10|
|powershell|ms-vscode|2022.10.0|
|python|ms-python|2022.16.1|
|ruby|rebornix|0.28.1|
|vscode-back-forward-button|grimmer|0.1.6|
|vscode-fileutils|sleistner|3.5.0|
|vscode-java-debug|vscjava|0.45.0|
|vscode-java-dependency|vscjava|0.21.0|
|vscode-java-pack|vscjava|0.25.3|
|vscode-java-test|vscjava|0.37.1|
|vscode-jupyter-cell-tags|ms-toolsai|0.1.6|
|vscode-jupyter-slideshow|ms-toolsai|0.1.5|
|vscode-maven|vscjava|0.39.0|
|vscode-pylance|ms-python|2022.10.40|
|vscode-ruby|wingrunr21|0.28.0|
|vscodeintellicode|VisualStudioExptTeam|1.2.29|;

</details>

### PR DESCRIPTION
**Pull request by Thaina**
fix #5433

### CODE REVIEW

**Comment by Thaina** [comment]
@microsoft-github-policy-service agree [company=""]

**Comment by Thaina** [comment]
@microsoft-github-policy-service agree

**Comment by Thaina** [comment]
@microsoft-github-policy-service agree

File: src/omnisharp/server.ts [hunk 1]
```diff
@@ -693,6 +686,15 @@ export class OmniSharpServer {
 
         const promise = new Promise<void>((resolve, reject) => {
             let listener: Disposable;
+            
+            serverProcess.stderr.on('data', (data: Buffer) => {
+                let trimData = removeBOMFromBuffer(data);
+                if (trimData.length > 0) {
+                    const err = trimData.toString();
+                    this._fireEvent(Events.StdErr, err);
+                    reject(new Error(err));
```
[the remark this answers is not in the document]
Comment (Thaina) [reply src/omnisharp/server.ts:695]: But whenever there really is an error in server the omnisharp vscode will just stop working silently

I think it safe to assume that error is really error. If there are log that not really error it then should be changed to a warning

Or are there any error handling mechanism more reliable from omnisharp server ?

**Comment by Rizary** [comment]
@Thaina is this the same issue where I got this on my OmniSharp log?
```
[fail]:  OmniSharp.MSBuild.Discovery.Providers.SdkOverrideInstanceProvider
        The Sdk path specified in the OmniSharp settings does not exist. Configured path is ''C:\Program Files\dotnet\sdk\7.0.102\''. Please update your settings and restart OmniSharp.
```

**Comment by Thaina** [comment]
I thought so. Any doubling `'` are jamming parsing of path

**Comment by Thaina** [comment]
It seem the main problem was fixed

**Pull request closed without merging**